libultraship/libultraship/SohImGuiImpl.cpp

```diff
@@ -401,116 +401,118 @@ namespace SohImGui {
+    void EnhancementSliderFloat(const char* text, const char* id, const char* cvarName, float min, float max, const char* format, float defaultValue, bool isPercentage)
     {
-        float val = CVar_GetFloat(cvarName.c_str(), defaultValue);
+        float val = CVar_GetFloat(cvarName, defaultValue);
 
         if (!isPercentage)
-            ImGui::Text(text.c_str(), val);
+            ImGui::Text(text, val);
         else
-            ImGui::Text(text.c_str(), static_cast<int>(100 * val));
+            ImGui::Text(text, static_cast<int>(100 * val));
 
-        if (ImGui::SliderFloat(id.c_str(), &val, min, max, format.c_str()))
+        if (ImGui::SliderFloat(id, &val, min, max, format))
         {
-            CVar_SetFloat(cvarName.c_str(), val);
+            CVar_SetFloat(cvarName, val);
             needs_save = true;
         }
 
         if (val < min)
         {
             val = min;
-            CVar_SetFloat(cvarName.c_str(), val);
+            CVar_SetFloat(cvarName, val);
             needs_save = true;
         }
 
         if (val > max)
         {
             val = max;
-            CVar_SetFloat(cvarName.c_str(), val);
+            CVar_SetFloat(cvarName, val);
             needs_save = true;
         }
     }
 
-    void EnhancementColor3(std::string text, std::string cvarName, float defaultColors[3])
+    void EnhancementColor3(const char* text, const char* cvarName, float defaultColors[3])
     {
-        int r = CVar_GetS32((cvarName + "_Red").c_str(), (defaultColors[0] * 255.0f));
-        int g = CVar_GetS32((cvarName + "_Green").c_str(), (defaultColors[1] * 255.0f));
-        int b = CVar_GetS32((cvarName + "_Blue").c_str(), (defaultColors[2] * 255.0f));
+
+        int r = CVar_GetS32((std::string(cvarName) + "_Red").c_str(), (int32_t)(defaultColors[0] * 255.0f));
+        int g = CVar_GetS32((std::string(cvarName) + "_Green").c_str(), (int32_t)(defaultColors[1] * 255.0f));
+        int b = CVar_GetS32((std::string(cvarName) + "_Blue").c_str(), (int32_t)(defaultColors[2] * 255.0f));
 
         float colors[3];
         colors[0] = r / 255.0f;
         colors[1] = g / 255.0f;
         colors[2] = b / 255.0f;
 
         {
-            if (ImGui::ColorEdit3(text.c_str(), colors))
+            if (ImGui::ColorEdit3(text, colors))
             {
-                CVar_SetS32((cvarName + "_Red").c_str(), (int)(colors[0] * 255));
-                CVar_SetS32((cvarName + "_Green").c_str(), (int)(colors[1] * 255));
-                CVar_SetS32((cvarName + "_Blue").c_str(), (int)(colors[2] * 255));
+                CVar_SetS32((std::string(cvarName) + "_Red").c_str(), (int)(colors[0] * 255));
+                CVar_SetS32((std::string(cvarName) + "_Green").c_str(), (int)(colors[1] * 255));
+                CVar_SetS32((std::string(cvarName) + "_Blue").c_str(), (int)(colors[2] * 255));
```

Comment:
In this case I think you're making it worse, since you create an identical string object containing cvarName 6 times here.